### PR TITLE
Catch Exception instead of using bare except.

### PR DESCRIPTION
**Pull request by maxzheng**
```
:!python3 /tmp/byte-of-python/programs/more_decorator.py
Write to a database or make a network call or etc.
This will be automatically retried if exception is thrown.
ERROR:retry:Attempt 1/5 failed : (('Some bad value',), {})
Traceback (most recent call last):
  File "/tmp/byte-of-python/programs/more_decorator.py", line 15, in wrapper_function
    return f(*args, **kwargs)
  File "/tmp/byte-of-python/programs/more_decorator.py", line 37, in save_to_database
    raise ValueError(arg)
ValueError: Some bad value
Write to a database or make a network call or etc.
This will be automatically retried if exception is thrown.

Press ENTER or type command to continue
```

### CODE REVIEW

File: programs/more_decorator.py [hunk 1]
```diff
@@ -24,18 +25,15 @@ def wrapper_function(*args, **kwargs):
...
 def save_to_database(arg):
     print("Write to a database or make a network call or etc.")
     print("This will be automatically retried if exception is thrown.")
-    global counter
-    counter += 1
+
     # This will throw an exception in the first call
     # And will work fine in the second call (i.e. a retry)
-    if counter < 2:
+    if not hasattr(save_to_database, 'called'):
```
Comment (swaroopch) [inline programs/more_decorator.py]: I'm not sure this is a good change, it distracts from the example's focus on decorators?

Comment (maxzheng) [reply programs/more_decorator.py]: It is less line of code than the global counter (global = generally bad), so to me, it feels less distracting? But if you insist, I can revert that part of it. Let me know.

Comment (swaroopch) [reply programs/more_decorator.py]: @maxzheng Can you please check if we have already discussed `hasattr` in the book before this section? If yes, let's keep it. If no, can you please revert that part? Thanks!

Comment (maxzheng) [reply programs/more_decorator.py]: Interestingly there is no hasattr use. Reverted for now. It is a fairly useful function, I would suggest adding that in the future.

**Comment by swaroopch** [comment]
Thanks!

**Pull request merged**